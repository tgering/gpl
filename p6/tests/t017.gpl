string window_title = "p6 t017.gpl -- compare to t017.jpg";

pixmap photo1(x = 100, y = 100, filename = "tests/1.bmp");
pixmap photo2(x = 100, y = 300, filename = "tests/2.bmp");
pixmap photo3(x = 300, y = 300, filename = "tests/3.bmp");
pixmap photo4(x = 300, y = 100, filename = "tests/4.bmp");

textbox title(x = 10, y = 10, text = "Four pixmaps");
