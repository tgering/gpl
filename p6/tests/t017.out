gpl.cpp::main()
  input file(tests/t017.gpl)
  random seed(42)
  read_keypresses_from_standard_input(false)
  dump_pixels(true, file = results/t017.mypixels)
  symbol_table(true)
  graphics(true)

gpl.cpp::main() Calling yyparse()


gpl.cpp::main() after call to yyparse().


Printing the symbol table from main()
pixmap photo1
{
    animation_block = NULL,
    blue = double(0.5),
    drawing_order = int(0),
    filename = string("tests/1.bmp"),
    green = double(0.5),
    h = int(10),
    proximity = int(4),
    red = double(0.5),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(10),
    x = int(100),
    y = int(100)
}
pixmap photo2
{
    animation_block = NULL,
    blue = double(0.5),
    drawing_order = int(0),
    filename = string("tests/2.bmp"),
    green = double(0.5),
    h = int(10),
    proximity = int(4),
    red = double(0.5),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(10),
    x = int(100),
    y = int(300)
}
pixmap photo3
{
    animation_block = NULL,
    blue = double(0.5),
    drawing_order = int(0),
    filename = string("tests/3.bmp"),
    green = double(0.5),
    h = int(10),
    proximity = int(4),
    red = double(0.5),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(10),
    x = int(300),
    y = int(300)
}
pixmap photo4
{
    animation_block = NULL,
    blue = double(0.5),
    drawing_order = int(0),
    filename = string("tests/4.bmp"),
    green = double(0.5),
    h = int(10),
    proximity = int(4),
    red = double(0.5),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(10),
    x = int(300),
    y = int(100)
}
textbox title
{
    animation_block = NULL,
    blue = double(0),
    drawing_order = int(0),
    green = double(0),
    h = int(11),
    proximity = int(4),
    red = double(0),
    size = double(0.1),
    space = int(10),
    text = string("Four pixmaps"),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(10),
    x = int(10),
    y = int(10)
}
string window_title = "p6 t017.gpl -- compare to t017.jpg"
gpl.cpp::main() Calling window->initialize().


gpl.cpp::main() Passing control to window->main_loop().

User quit program. Printing the symbol table.
pixmap photo1
{
    animation_block = NULL,
    blue = double(0.5),
    drawing_order = int(0),
    filename = string("tests/1.bmp"),
    green = double(0.5),
    h = int(113),
    proximity = int(4),
    red = double(0.5),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(75),
    x = int(100),
    y = int(100)
}
pixmap photo2
{
    animation_block = NULL,
    blue = double(0.5),
    drawing_order = int(0),
    filename = string("tests/2.bmp"),
    green = double(0.5),
    h = int(100),
    proximity = int(4),
    red = double(0.5),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(150),
    x = int(100),
    y = int(300)
}
pixmap photo3
{
    animation_block = NULL,
    blue = double(0.5),
    drawing_order = int(0),
    filename = string("tests/3.bmp"),
    green = double(0.5),
    h = int(100),
    proximity = int(4),
    red = double(0.5),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(150),
    x = int(300),
    y = int(300)
}
pixmap photo4
{
    animation_block = NULL,
    blue = double(0.5),
    drawing_order = int(0),
    filename = string("tests/4.bmp"),
    green = double(0.5),
    h = int(100),
    proximity = int(4),
    red = double(0.5),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(133),
    x = int(300),
    y = int(100)
}
textbox title
{
    animation_block = NULL,
    blue = double(0),
    drawing_order = int(0),
    green = double(0),
    h = int(11),
    proximity = int(4),
    red = double(0),
    size = double(0.1),
    space = int(10),
    text = string("Four pixmaps"),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(97),
    x = int(10),
    y = int(10)
}
string window_title = "p6 t017.gpl -- compare to t017.jpg"
